ros/processor/insights_engine_result_consumer.py

```diff
@@ -116,6 +130,26 @@ def process_report(self, host, reports):
                     number_of_recommendations=rec_count,
                     state=SYSTEM_STATES[state_key]
                 )
+                LOG.info(
+                    f"{self.prefix} - System created/updated successfully: {host['id']}"
+                )
+
+                performance_utilization = {
+                    'memory': system_details['mem_utilization'],
+                    'cpu': system_details['cpu_utilization'],
+                    'io': system_details['io_utilization']
+                }
```

Comment:
Can you point me to the logic where you're handling the Waiting for Data state i.e. the 0 day experience where you won't have any actual metric values available?

Reply:
For such a case, won't the engine msg have 0 values? I'm not too sure

Reply:
According to the older workflow, we were getting `NO_PCP_DATA` key hit from the rules and we had to handle it in our processor to store data in the db with no value scenario. I am guessing that should be the case even now, but I can confirm once I see the actual message that will come from the insights rule.

Reply:
I noticed that portion of the code and had removed it intentionally.
IMO we should receive the 0 values from the advisor for utilization in case its a fresh system.
Yes, lets wait for the rules to go live and then take a call, thanks!

Reply:
@saltgen As per Xiangce's latest comment on CEECBA-5875, no metadata would be returned. Hence, we will have to set `*_utilization` values to 0 as present in the red diff [here](https://github.com/RedHatInsights/ros-backend/pull/138/files#diff-3ec10f47264a06a134d0bf1ce745fb26ad086de0af6da0cf6e83c00dcf2c4193L156). We should check if the for the NO_PCP_DATA condition and accordingly set the values.

Reply:
I see, okay, I will make the changes and push shortly.

Reply:
As discussed, awaiting live rules and archives to test changes.
Will be adding suggested change in one go.

Reply:
Also, you will need to handle the Optimized state here too, otherwise it could give an error as follows:
```python
2021-12-14 13:18:10,504 - INFO  - process_report - PROCESSING ENGINE RESULTS - No ROS rule hits found for system with inventory id: cd002de7-851e-40af-a70e-24554ab10b55. Hence, marking the state as Optimized.
2021-12-14 13:18:10,506 - INFO  - process_report - PROCESSING ENGINE RESULTS - System created/updated successfully: cd002de7-851e-40af-a70e-24554ab10b55
2021-12-14 13:18:10,506 - ERROR  - process_report - PROCESSING ENGINE RESULTS - Unable to add host ip-172-31-45-168.ap-south-1.compute.internal to DB belonging to account: 0000001 - 'mem_utilization'

```

Reply:
> @saltgen As per Xiangce's latest comment on CEECBA-5875, no metadata would be returned. Hence, we will have to set `*_utilization` values to 0 as present in the red diff [here](https://github.com/RedHatInsights/ros-backend/pull/138/files#diff-3ec10f47264a06a134d0bf1ce745fb26ad086de0af6da0cf6e83c00dcf2c4193L156). We should check for the NO_PCP_DATA condition and accordingly set the values.

Added zero_value initialization for NO_PCP_DATA. Changes pushed :rocket: 



> Also, you will need to handle the Optimized state here too, otherwise it could give an error as follows:
> 
> ```python
> 2021-12-14 13:18:10,504 - INFO  - process_report - PROCESSING ENGINE RESULTS - No ROS rule hits found for system with inventory id: cd002de7-851e-40af-a70e-24554ab10b55. Hence, marking the state as Optimized.
> 2021-12-14 13:18:10,506 - INFO  - process_report - PROCESSING ENGINE RESULTS - System created/updated successfully: cd002de7-851e-40af-a70e-24554ab10b55
> 2021-12-14 13:18:10,506 - ERROR  - process_report - PROCESSING ENGINE RESULTS - Unable to add host ip-172-31-45-168.ap-south-1.compute.internal to DB belonging to account: 0000001 - 'mem_utilization'
> ```

As discussed on Slack since metadata is unavailable only in-case of `NO_PCP_DATA`, letting this be :v: